File: php/binary-tree-inorder-traversal.php
<?php

class Node {
    public $val;
    public $left;
    public $right;

    public function __construct($val) {
        $this->val = $val;
        $this->left = null;
        $this->right = null;
    }
}

class Solution {
    public $result = [];

    public function inorderTraversal($root) {
        $stack = [];
        $current = $root;
        while(!is_null($current) || count($stack) > 0) {
            if(!is_null($current)) {
                array_push($stack, $current);
                $current = $current->left;
            } else {
                $current = array_pop($stack);
                array_push($this->result, $current->val);
                $current = $current->right;
            }
        }
        return $this->result;
    }
}

$root = new Node(1);
$root->right = new Node(2);
$root->right->left = new Node(3);

$solution = new Solution();
print_r($solution->inorderTraversal($root));

File: python/binary-tree-inorder-traversal.py
class Node:
    def __init__(self, val):
        self.val = val
        self.left = None
        self.right = None

class Solution:
    def inorderTraversal(self, root):
        if root is None:
            return []
        stack = []
        res = []
        while root or stack:
            while root:
                stack.append(root)
                root = root.left
            if stack:
                node = stack.pop()
                res.append(node.val)
                root = node.right
        return res
    
root = Node(1)
root.right = Node(2)
root.right.left = Node(3)
print(Solution().inorderTraversal(root))